### ISSUE TITLE
Multiblock - Text Blocks appear to duplicate at times

### ISSUE THREAD

**Issue by jprovasnik** [catalyst/moodle-block_multiblock#108]
I'm using Moodle 4.1 Workplace and I notice that Multiblock textblocks appear to 'duplicate' at times when I have a bunch of Text Blocks (so there are 2 of each text block) -- ever seen this before? Just trying to figure out what I (or my site) might be doing to cause this, or if a scheduled task is somehow contributing. Happens to blocks in both Tabs and Accordion. I know Workplace isn't something that developers support, so I really just wanted to ask if this has been seen before in regular Moodle instances. 

![dupliate](https://github.com/catalyst/moodle-block_multiblock/assets/44345287/e71d0a64-73f5-4672-9f00-8ca737000ddd)

**Comment by danmarsden** [comment]
Thanks for the report - I haven't heard of anyone mentioning this before - I assume this isn't a display-only issue, and when you edit the block you see the duplicates there as well? - if so, I'd be suspicious of a restore or import process that is causing the duplicates - did you create this course from a template course or use an import process to restore something within the course?

**Comment by jprovasnik** [comment]
Yes, there are duplicate of the text blocks as well - half from back in November when they were added, and the other half appeared this morning. There are on a Moodle Workplace "Custom Page" so they live outside of the courses, but I'll look through the Logs and Scheduled Task Logs to see if there are any copy/import processes that ran around that time frame.

**Comment by jprovasnik** [comment]
Ok, it coincided with a Log entry for when another workplace Custom Page was created. Will see if I can replicate this but at least I can line it up with an event now.

**Comment by jprovasnik** [comment]
Ok - It's not from "adding" a new Custom Page, but instead it happens when "Duplicating" a Content Page that contains Multiblock, then the original Content page's Multiblock contents gets duplicated. So just what you had suspected. Ok, I know you don't work with Workplace, but I just wanted to report this. thanks!

**Comment by danmarsden** [comment]
We do "work" with workplace (Catalyst is a Premium Moodle Partner), but I don't think we've had this reported with a workplace site yet - thanks for the report and the details on how to reproduce it though!

If you have internal development capabilities, feel free to submit a pull request (or ask your Moodle Partner to investigate and send us one.)

**Comment by danmarsden** [comment]
this is now fixed in the latest branch - quite a tricky one to prevent on the multiblock side - but we found a workaround which appears to fix it which is now in the core release!
